src/App.tsx

```diff
@@ -1,26 +1,53 @@
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-import { Loader, TodoFilter, TodoList, TodoModal } from './components';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { TodoModal } from './components/TodoModal';
+import { Loader } from './components/Loader';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { actions as todosActions } from './features/todos';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { filterTodos } from './utils/todos';
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+
+  const todos = useAppSelector(state => state.todos);
+  const { query, status } = useAppSelector(state => state.filter);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then((data: Todo[]) => dispatch(todosActions.setTodos(data)))
+      .finally(() => setIsLoading(false));
```

Comment:
Consider adding error handling for the `getTodos` API call. You can include a `.catch()` block to handle any errors that might occur during the fetch operation.